CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+option(SANITIZE_ADDRESS "Address Sanitizer." OFF)
+
+if (SANITIZE_ADDRESS)
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+endif()
```

Comment:
These don't really belong in a project CML. These should be externalized to a preset, like [this](https://github.com/friendlyanon/cmake-init-header-only/blob/master/CMakePresets.json#L46) and [this](https://github.com/friendlyanon/cmake-init-header-only/blob/master/CMakePresets.json#L106).

Reply:
I'll admit it was a mistake to make them global in this case.
Wouldn't this do the job? Since CXX 20 is required for this header. Haven't worked with presets will take a look at that option too.

set_property(TARGET ${This} PROPERTY C_STANDARD 99)
set_property(TARGET ${This} PROPERTY CXX_STANDARD 20)
if (SANITIZE_ADDRESS)
set_property (TARGET ${This} PROPERTY CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
set_property (TARGET ${This} PROPERTY CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
endif()

Reply:
CMake has [compile features](https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards) to indicate standard requirement. Everything else is an external concern, a social requirement if you will. Project CMLs should only be concerned with build and usage requirements.